cypress/pages/AgodaPage.ts

```diff
@@ -0,0 +1,70 @@
+
+import AgodaSelectors from "../selectors/agodaSelectors";
+//import { select } from '@testing-library/dom'; // Assuming usage of @testing-library/dom
+
+class AgodaPage {
+  visit(): void {
+    cy.visit('https://www.agoda.com');
+  }
+
+  enterLocation(location: string): void {
```

Comment:
Move common methods into basepage and all pages should inherit it and reuse same functionlity i.e click, enter, getText etc..

Reply:
and add validation in each function if specific action is doing fine or not. i.e check if 'click' is done.